shop/shop-service/src/main/java/shop/woowasap/shop/service/CartConnectorService.java

```diff
@@ -28,4 +29,9 @@ public Optional<Cart> findByCartIdAndUserId(final long cartId, final long userId
             return Optional.empty();
         }
     }
+
+    @Override
+    public void clearCartByUserId(final long userId) {
+        cartService.clearCartByUserId(userId);
```

Comment:
CartConnectorService에서 CartService를 호출하기보다, 
cartRepository를 호출하는 방식으로 변경하는건 어떨까요? 

cartService는 현재 web client와 상호작용 하고 있고, cartConnectorService는 다른 모듈과 상호작용 하고있는데, `cartConnectorService -> cartService` 처럼 참조하면, web client의 변경요청이 다른 모듈로 이어질수도 있을거 같아요

```suggestion
	Cart cart = cartRepository.getByUserId(userId);
	cart.clear();
	cartRepository.persist(cart);
```

Reply:
오 cart안의 cartProduct 콜렉션을 비우면 되군요....!😅 수정하겠습니당

Reply:
userId에 해당하는 cart가 없는 경우,새로운 id를 생성해서 카트를 생성해줘야하는데, 이 역할이 CartService의 역할이라고 생각해서 CartService를 호출했습니다

Reply:
clearCartByUserId가 cart에 있는 아이템을 구매한 직후에 호출되는것 아닌가요?
맞다면, 항상 cart가 존재할것 같아요~

`CartConnectorService`와 `CartService` 모두 애플리케이션 플로우를 담당한다는 점에서 역할은 비슷한데요, 다른점은 `서로 다른 변경이유를 갖고있다`는것 이라고 생각합니다.

예를들어, CartConnectorService는 자신을 사용하는 모듈에 의해서 변경이 일어나지만, CartService는 클라이언트에 의해서 변경이 발생하죠

따라서

> userId에 해당하는 cart가 없는 경우,새로운 id를 생성해서 카트를 생성해줘야하는데

와 같은 로직이 필요하다면, CartConnectorService에 두는게 맞다고 생각됩니다.

로직이 중복되는것이 걱정이라면, 두 로직이 진짜 중복인지, 가짜중복인지, 항상 같은 방향으로 발전하는지 고민해보면 좋을거 같아요

Reply:
새로운 Cart 가 생성되는 것만 생각하고,  CartConnectorService 와 CartService 각각의 변경 이유에 대해서는 생각해보지 못한 것 같네요..! CartConnector에서 CartService의 의존성을 제거하도록 수정했습니당